providers/implementations/signature/dsa_sig.c

```diff
@@ -156,14 +156,20 @@ static int dsa_setup_md(PROV_DSA_CTX *ctx,
                                "%s exceeds name buffer length", mdname);
             goto err;
         }
+        /* XOF digests don't work */
+        if ((EVP_MD_get_flags(md) & EVP_MD_FLAG_XOF) != 0) {
+            ERR_raise(ERR_LIB_PROV, PROV_R_XOF_DIGESTS_NOT_ALLOWED);
+            goto err;
+        }
 #ifdef FIPS_MODULE
         {
             int sha1_allowed = (ctx->operation != EVP_PKEY_OP_SIGN);
 
             if (!ossl_fips_ind_digest_sign_check(OSSL_FIPS_IND_GET(ctx),
                                                  OSSL_FIPS_IND_SETTABLE1,
                                                  ctx->libctx, md_nid, sha1_allowed,
-                                                 desc))
+                                                 desc,
+                                                 &FIPS_fips_signature_digest_check))
```

Comment:
Is the `&` operator necessary here, and elsewhere? `FIPS_fips_signature_digest_check` is already a function, so should already be a pointer?

Reply:
There is going to be a pass over these after all the changes are in, to change the names to something more meaningful, this should be resolved at that time.

Reply:
The `&` is just a habit of mine.  I find it more readable and it highlights that a function pointer is being passed.
The coding style is mum on including it or not.

Reply:
I'm not sure we need to rename these -- they are inside the FIPS provider which can only be built as a shared object and the namespace is kept completely internal.  I.e. they cannot pollute the calling application's namespace.

Reply:
It was more about putting 'config' in the name

Reply:
Perhaps the coding standard should be updated to resolve this ambiguity.